bindings/python/unicorn/unicorn.py

```diff
@@ -1,492 +1,303 @@
-        if dll in _loaded_windows_dlls:
-            continue
+def __load_uc_lib() -> ctypes.CDLL:
+    from pathlib import Path, PurePath
 
-        lib_file = os.path.join(path, dll)
-        if ('/' not in path and '\\' not in path) or os.path.exists(lib_file):
-            try:
-                #print('Trying to load Windows library', lib_file)
-                ctypes.cdll.LoadLibrary(lib_file)
-                #print('SUCCESS')
-                _loaded_windows_dlls.add(dll)
-            except OSError as e:
-                #print('FAIL to load %s' %lib_file, e)
-                continue
-
-# Initial attempt: load all dlls globally
-if sys.platform in ('win32', 'cygwin'):
-    _load_win_support('')
-
-def _load_lib(path, lib_name):
-    try:
-        if sys.platform in ('win32', 'cygwin'):
-            _load_win_support(path)
-
-        lib_file = os.path.join(path, lib_name)
-        dll = ctypes.cdll.LoadLibrary(lib_file)
-        #print('SUCCESS')
-        return dll
-    except OSError as e:
-        #print('FAIL to load %s' %lib_file, e)
-        return None
-
-_uc = None
-
-# Loading attempts, in order
-# - user-provided environment variable
-# - pkg_resources can get us the path to the local libraries
-# - we can get the path to the local libraries by parsing our filename
-# - global load
-# - python's lib directory
-# - last-gasp attempt at some hardcoded paths on darwin and linux
-
-_path_list = [os.getenv('LIBUNICORN_PATH', None),
-              pkg_resources.resource_filename(__name__, 'lib'),
-              os.path.join(os.path.split(__file__)[0], 'lib'),
-              '',
-              distutils.sysconfig.get_python_lib(),
-              "/usr/local/lib/" if sys.platform == 'darwin' else '/usr/lib64',
-              os.getenv('PATH', '')]
-
-#print(_path_list)
-#print("-" * 80)
-
-for _path in _path_list:
-    if _path is None: continue
-    _uc = _load_lib(_path, _lib.get(sys.platform, "libunicorn.so"))
-    if _uc is not None:
-        break
-
-# Try to search old unicorn1 library without SONAME
-if _uc is None:
-    for _path in _path_list:
-        if _path is None:
-            continue
-        
-        _uc = _load_lib(_path, "libunicorn.so")
-        if _uc is not None:
-            # In this case, show a warning for users
-            print("Found an old style dynamic library libunicorn.so, consider checking your installation", file=sys.stderr)
-            break
-
-if _uc is None:
-    raise ImportError("ERROR: fail to load the dynamic library.")
-
-__version__ = "%u.%u.%u" % (uc.UC_VERSION_MAJOR, uc.UC_VERSION_MINOR, uc.UC_VERSION_PATCH)
-
-# setup all the function prototype
-def _setup_prototype(lib, fname, restype, *argtypes):
-    try:
-        getattr(lib, fname).restype = restype
-        getattr(lib, fname).argtypes = argtypes
-    except AttributeError:
-        raise ImportError("ERROR: Fail to setup some function prototypes. Make sure you have cleaned your unicorn1 installation.")
-
-ucerr = ctypes.c_int
```

Comment:
Though these alias doesn't make too much sense, I think it doesn't matter too much to just leave them there ;).

Reply:
They were relocated to somewhere else to help with code readability.